core/src/main/java/lucee/runtime/db/QoQ.java

```diff
@@ -30,33 +35,51 @@
 import lucee.runtime.exp.DatabaseException;
 import lucee.runtime.exp.PageException;
 import lucee.runtime.op.Caster;
+import lucee.runtime.op.Decision;
 import lucee.runtime.op.Operator;
+import lucee.runtime.sql.QueryPartitions;
 import lucee.runtime.sql.Select;
 import lucee.runtime.sql.SelectParser;
 import lucee.runtime.sql.Selects;
 import lucee.runtime.sql.exp.BracketExpression;
 import lucee.runtime.sql.exp.Column;
 import lucee.runtime.sql.exp.ColumnExpression;
 import lucee.runtime.sql.exp.Expression;
+import lucee.runtime.sql.exp.Literal;
 import lucee.runtime.sql.exp.op.Operation;
 import lucee.runtime.sql.exp.op.Operation1;
 import lucee.runtime.sql.exp.op.Operation2;
 import lucee.runtime.sql.exp.op.Operation3;
+import lucee.runtime.sql.exp.op.OperationAggregate;
 import lucee.runtime.sql.exp.op.OperationN;
 import lucee.runtime.sql.exp.value.Value;
 import lucee.runtime.sql.exp.value.ValueNumber;
+import lucee.runtime.type.Array;
 import lucee.runtime.type.ArrayImpl;
 import lucee.runtime.type.Collection;
 import lucee.runtime.type.Collection.Key;
+import lucee.runtime.type.KeyImpl;
 import lucee.runtime.type.Query;
 import lucee.runtime.type.QueryColumn;
+import lucee.runtime.type.QueryColumnImpl;
 import lucee.runtime.type.QueryImpl;
+import lucee.runtime.type.util.ArrayUtil;
 
 /**
  * 
  */
 public final class QoQ {
+	final static private Collection.Key paramKey = new KeyImpl("?");
```

Comment:
@michaeloffner Sorry, I don't follow what your question is.  I created the Key here just to prevent the overheard of creating it in every place it was used.  We can put this in `KeyConstants.java`, but I wasn't sure what the criteria was for doing that.

Reply:
It's fine as it is, i was just not sure about the naming convention.